giotto/mapper/nerve.py

```diff
@@ -75,11 +75,7 @@ def fit(self, X, y=None):
 
         """
         # TODO: Include a validation step for X
-        self.X_ = reduce(iconcat, X, [])
-        # Preprocess X by 1) flattening and 2) extending each tuple
-        self.X_ = [(node_info[0], *node_info[1])
-                   for node_info in zip(range(len(self.X_)), self.X_)]
-        self.edges_ = list(self._generate_edges(self.X_))
+        self.X_, self.edges_ = self._graph_data_creation(X)
```

Comment:
I would prefer it if the utility function was called _create_graph_data() since verbs are better for revealing intent